python/fledge/plugins/south/person_detection/videostream.py

```diff
@@ -44,23 +45,42 @@ class VideoStream:
     """ Camera object that controls video streaming from the Camera
     """
 
-    def __init__(self, resolution=(640, 480), framerate=30, source=0, enable_thread=False):
+    def __init__(self, resolution=(640, 480), framerate=30, source=0, enable_thread=False, stream_url="None",
+                 stream_protocol="udp", opencv_backend="ffmpeg"):
         # Initialize the PiCamera and the camera image stream
 
-        if detect_mjpg_camera(source):
-            # only mjpg  pixel format and coral camera are supported.
-            self.stream = cv2.VideoCapture(source)
-            _ = self.stream.set(cv2.CAP_PROP_FOURCC, cv2.VideoWriter_fourcc(*'MJPG'))
-            _ = self.stream.set(3, resolution[0])
-            _ = self.stream.set(4, resolution[1])
-        elif detectCoralDevBoard():
-            self.stream = cv2.VideoCapture(source)
-            # cannot change camera resolution on coral device through opencv API.
+        if stream_url is not None:
+            if opencv_backend == 'ffmpeg':
+                if stream_protocol == "udp":
+                    os.environ["OPENCV_FFMPEG_CAPTURE_OPTIONS"] = "rtsp_transport;udp"
+                    self.stream = cv2.VideoCapture(stream_url, cv2.CAP_FFMPEG)
+
+                # not implementing tcp for now for that Need to create stream at tcp.
+                else:
+                    _LOGGER.error("Only streams using udp protocol are being processed for now. Need a "
+                                  "way to send stream via tcp. For more information refer to documentation")
+                    raise NotImplementedError
+
+            # not implementing other backend like gstreamer though opencv supports them as well.
+            else:
+                _LOGGER.error("Only ffmpeg is implemented for now.")
+                raise NotImplementedError
+
         else:
-            # If the device is not supported self.stream would be None.
-            # There could be another format such as YUYV which may support changing camera resolution
-            # through Open CV API. Omitting these calls for now.
-            self.stream = cv2.VideoCapture(source)
+            if detect_mjpg_camera(source):
+                # only mjpg  pixel format and coral camera are supported.
```

Comment:
Last 'else' case is catch all, it supports other formats also.

Reply:
https://github.com/fledge-iot/fledge-south-person-detection/pull/17#discussion_r597619731